garak/probes/base.py

```diff
@@ -183,6 +183,10 @@ def _mint_attempt(
         self, prompt=None, seq=None, notes=None, lang="*"
     ) -> garak.attempt.Attempt:
         """function for creating a new attempt given a prompt"""
+        if hasattr(self, "generator") and hasattr(self.generator, "system_prompt"):
+            system_prompt = self.generator.system_prompt
+        else:
+            system_prompt = None
```

Comment:
I'd like a different pattern for doing this. I guess we mint attempts with an adversarial prompt. that is represented as a conversation[] with one message. maybe sysprompts can get inserted at index 0, maybe in assembly (a hook in probe.base?)

-- secondary question: who owns the system prompt? is it global? (loading these in YAML is a pita) -- some probes are looking to add specific system prompts so they can check if they're accurately recovering the system prompt later. I'm thinking maybe they're something configured on a `generator` rather than `probe`,and that @jmartin-tech would be a good consult here